cli/collection_create.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"encoding/json"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	"github.com/sourcenetwork/defradb/client"
+)
+
+func MakeCollectionCreateCommand() *cobra.Command {
+	var file string
+	var cmd = &cobra.Command{
+		Use:   "create <document>",
+		Short: "Create a new document.",
+		Long: `Create a new document.
+
+Example: create document
+  defradb client collection create --name User '{ "name": "Bob" }'
+
+Example: create documents
+  defradb client collection create --name User '[{ "name": "Alice" }, { "name": "Bob" }]'
+		`,
+		Args: cobra.RangeArgs(0, 1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			col, ok := cmd.Context().Value(colContextKey).(client.Collection)
```

Comment:
suggestion: Similar to `store`, I'd suggest refactoring out `cmd.Context().Value(colContextKey).(client.Collection)` into a private func, located next to where it is set (collection.go).